samples: drivers: display: frdm_rw612 overlay references undefined st7796s label

## Describe the bug

The display sample fails to build for `frdm_rw612` because the
board-specific sample overlay references `&st7796s`, but that node label
is not defined by the `frdm_rw612` board DTS.

This breaks the `samples/drivers/display` twister job on main.

## To Reproduce

build on commit
0570f6d6bc10aa9d185ed4aca7660b40a082bc0b.

Reproducer from the report:

```sh
git clone https://github.com/zephyrproject-rtos/zephyr -br main
cd zephyr
scripts/twister -p frdm_rw612 -j 12 -T samples/drivers/display \
  --build-only --inline-logs -M all
```

## Actual behavior

Twister reports `CMake build failure - devicetree error` for both:

- `sample.display.builtin`
- `sample.display.shield`

Error excerpt:

```text
devicetree error:
.../samples/drivers/display/boards/frdm_rw612.overlay:11:
parse error: undefined node label 'st7796s'
```

## Expected behavior

`sample.display.builtin` and the shield-based display test should build
successfully, or the board-specific overlay should not be applied in a
way that references nodes unavailable on the base board.

## Root cause analysis

At commit 0570f6d6bc10aa9d185ed4aca7660b40a082bc0b:

- `samples/drivers/display/boards/frdm_rw612.overlay` contains:

```dts
&st7796s {
	zephyr,pm-device-runtime-auto;
};
```

- `boards/nxp/frdm_rw612/frdm_rw612.dts` and
  `boards/nxp/frdm_rw612/frdm_rw612_common.dtsi` do not define an
  `st7796s` node label.
- The label is defined by the shield overlay
  `boards/shields/lcd_par_s035/lcd_par_s035_spi.overlay`.

So the sample's board overlay now assumes a display controller node that
is not present on the base board and is only introduced by a specific
shield overlay.

This appears to have been introduced by commit
`f0515290e6b5b06a1023deb89249713db363c6b3`
("samples: display: enable runtime PM for frdm_rw612"), which added the
`&st7796s` block to `samples/drivers/display/boards/frdm_rw612.overlay`.

## Additional context

The previous FRDM_RW612 sample overlay commit
`f5691120cad86fffa776d7abf801d5ae89c3f29c`
("samples: display: Add overlay for FRDM_RW612 boards") only enabled
`&standby` and did not reference `st7796s`.

A likely fix is to avoid referencing `&st7796s` from the generic
`frdm_rw612` sample overlay, and move the runtime PM property to a
shield-specific overlay or otherwise gate it so it is only applied when
the shield-provided `st7796s` node exists.

> Hi [@jacob-wienecke-nxp](https://github.com/jacob-wienecke-nxp) , could you please check whether boards/shields/lcd_par_s035/boards/frdm_rw612.overlay is the better place for this code? thanks.
> 
> [zephyr/samples/drivers/display/boards/frdm_rw612.overlay](https://github.com/zephyrproject-rtos/zephyr/blob/0570f6d6bc10aa9d185ed4aca7660b40a082bc0b/samples/drivers/display/boards/frdm_rw612.overlay#L11-L13)
> 
> Lines 11 to 13 in [0570f6d](/zephyrproject-rtos/zephyr/commit/0570f6d6bc10aa9d185ed4aca7660b40a082bc0b)
> 
>  &st7796s { 
>  	zephyr,pm-device-runtime-auto; 
>  };

@hakehuang 
@zejiang0jason 

I'd prefer to keep it in the sample overlay.

`zephyr,pm-device-runtime-auto` is only needed because this sample enables runtime PM: `CONFIG_PM` / `CONFIG_PM_DEVICE_RUNTIME` / `CONFIG_PM_DEVICE_SYSTEM_MANAGED` are set in [`samples/drivers/display/boards/frdm_rw612.conf`](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/display/boards/frdm_rw612.conf). That's a sample-local decision. `boards/shields/lcd_par_s035/boards/frdm_rw612.overlay` is applied to *every* sample that uses this shield on FRDM-RW612, including ones that don't enable runtime PM, so putting the property there couples a PM-runtime tuning to the shield rather than to the config that actually requires it.

On FRDM-RW612, `CONFIG_PM_DEVICE_SYSTEM_MANAGED` is currently a hard requirement to resume the board's power domains; without it, nothing does. System-managed PM also drives individual *device* power transitions, which is the coupling RFC #84049 pushes back on. Since this board can't drop `SYSTEM_MANAGED`, the external `&st7796s` display has to opt into runtime PM explicitly via `zephyr,pm-device-runtime-auto`.

Note the sample already hard-depends on a display shield on this board. `lcd_par_s035_spi` is the out-of-the-box shield ([Shields for Supported Features](https://docs.zephyrproject.org/latest/boards/nxp/frdm_rw612/doc/index.html#shields-for-supported-features)).

Without a shield, it doesn't build regardless of this override; removing the `&st7796s` label from the sample overlay just changes *which* error you get:

```console
west build -p always -b frdm_rw612 samples/drivers/display
```
```
FAILED: CMakeFiles/app.dir/src/display.c.obj
zephyr/include/zephyr/device.h:96:41: error: '__device_dts_ord_DT_CHOSEN_zephyr_display_ORD' undeclared (first use in
this function)
zephyr/samples/drivers/display/src/display.c:299:23: note: in expansion of macro 'DEVICE_DT_GET'
  299 |         display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
ninja: build stopped: subcommand failed.
```
So keeping the override next to the sample's PM config, rather than in the shield, is deliberate: it avoids affecting any other sample that uses this shield.

While there is potential to add it to the shield board overlay location, validating that across the existing samples that can use this shield, as well as current user applications, was not in scope for PR #105421 and likely needs a migration guide.

--- a/boards/shields/lcd_par_s035/boards/frdm_rw612.overlay
+++ b/boards/shields/lcd_par_s035/boards/frdm_rw612.overlay
@@ -1,5 +1,5 @@
 /*
- * Copyright 2025 NXP
+ * Copyright 2025-2026 NXP
  *
  * SPDX-License-Identifier: Apache-2.0
  */
@@ -11,4 +11,8 @@
 	nxp,timer0-ratio = <15>;
 	/* Lower timer1 ratio to enable shorter TE delay */
 	nxp,timer1-ratio = <0>;
+};
+
+&st7796s {
+	zephyr,pm-device-runtime-auto;
 };

--- a/samples/drivers/display/boards/frdm_rw612.overlay
+++ b/samples/drivers/display/boards/frdm_rw612.overlay
@@ -1,13 +1,9 @@
 /*
- * Copyright 2025 NXP
+ * Copyright 2025-2026 NXP
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 &standby {
 	status = "okay";
-};
-
-&st7796s {
-	zephyr,pm-device-runtime-auto;
 };

